pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -54,32 +54,23 @@ func MaybeAddVolumeRefConfig(ctx *zedmanagerContext, appInstID uuid.UUID,
 	log.Functionf("MaybeAddVolumeRefConfig done for %s", key)
 }
 
-// MaybeRemoveVolumeRefConfig decreases the RefCount and deletes the VolumeRefConfig
-// when the RefCount reaches zero
+// MaybeRemoveVolumeRefConfig unpublishes VolumeRefConfig for volumemgr (AppInstanceConfig remains unchanged)
 func MaybeRemoveVolumeRefConfig(ctx *zedmanagerContext, appInstID uuid.UUID,
 	volumeID uuid.UUID, generationCounter, localGenerationCounter int64) {
 
-	key := fmt.Sprintf("%s#%d", volumeID.String(),
-		generationCounter+localGenerationCounter)
+	// construct the key as in VolumeRefConfig.Key()
```

Comment:
Key(), because in line 65 we use it to lookup VolumeRefConfig and not VolumeConfig. For looking up references we use their Key, for looking up respective volumes we use VolumeKey.